packages/wdio-browserstack-service/src/service.ts

```diff
@@ -0,0 +1,240 @@
+    private _failReasons: string[] = [];
+    private _scenariosThatRan: string[] = [];
+    private _preferScenarioName: boolean;
+    private _strict: boolean;
+    private _failureStatuses: string[] = ['failed', 'ambiguous', 'undefined', 'unknown'];
+    private _caps: Capabilities;
+    private _browser?: Browser;
+    private _fullTitle?: string;
+    constructor (options: BrowserstackConfig = {}, caps: Capabilities, private _config: WebdriverIO.Config) {
+        // Cucumber specific
+        this._preferScenarioName = Boolean(options.preferScenarioName)
+        this._strict = Boolean(_config.cucumberOpts && _config.cucumberOpts.strict)
+        // See https://github.com/cucumber/cucumber-js/blob/master/src/runtime/index.ts#L136
+        this._strict && this._failureStatuses.push('pending')
+        this._caps = caps
+    }
+
+    /**
+     * if no user and key is specified even though a browserstack service was
+     * provided set user and key with values so that the session request
+     * will fail
+     */
+    beforeSession (config: WebdriverIO.Config) {
+        if (!config.user) {
+            config.user = 'NotSetUser'
+        }
+
+        if (!config.key) {
+            config.key = 'NotSetKey'
+        }
+        this._config.user = config.user
+        this._config.key = config.key
+    }
+
+    before(caps: Capabilities, specs: string[], browser: Browser) {
+        this._browser = browser
+
+        // Ensure capabilities are not null in case of multiremote
+        const capabilities = this._browser.capabilities || {}
+        if (capabilities.app || this._caps.app) {
+            this._sessionBaseUrl = 'https://api-cloud.browserstack.com/app-automate/sessions'
+        }
+
+        this._scenariosThatRan = []
+
+        return this._printSessionURL()
+    }
+
+    beforeSuite (suite: WebdriverIO.Suite) {
+        this._fullTitle = suite.title
+        return this._updateJob({ name: this._fullTitle })
+    }
+
+    beforeFeature(uri: string, feature: Feature) {
+        this._fullTitle = feature.document.feature.name
+        return this._updateJob({ name: this._fullTitle })
+    }
+
+    afterTest(test: WebdriverIO.Test, context: Context, results: WebdriverIO.TestResult) {
+        const { error, passed } = results
+
+        this._fullTitle = (
+            /**
+             * Jasmine
+             */
+            test.fullName ||
+            /**
+             * Mocha
+             */
+            `${test.parent} - ${test.title}`
+        )
+
+        if (!passed) {
+            this._failReasons.push((error && error.message) || 'Unknown Error')
+        }
+    }
+
+    after(result: number) {
+        // For Cucumber: Checks scenarios that ran (i.e. not skipped) on the session
+        // Only 1 Scenario ran and option enabled => Redefine session name to Scenario's name
+        if (this._preferScenarioName && this._scenariosThatRan.length === 1){
```

Comment:
based on the suggestion above:
```suggestion
        if (this._options._preferScenarioName && this._scenariosThatRan.length === 1){
```

Reply:
Done! (I didn't commit this suggestion as "preferScenarioName" shouldn't be suffixed by underscore)